### PR TITLE
fix(device_info_plus): Fix deprecation warning on MacOS

### LINKED ISSUE #2376: [Bug]: Warrning on macOS for device_info_plus plugin
### Platform

macOS 14.1

### Plugin

device_info_plus

### Version

9.1.0

### Flutter SDK

3.13.9

### Steps to reproduce

1. Create a new project using `flutter create`
2. Add the plugin `device_info_plus` to the project
3. Run the project on macOS

### Code Sample

_No response_

### Logs

```shell
.pub-cache/hosted/pub.dev/device_info_plus-9.1.0/macos/Classes/SystemUUID.swift:7:72: warning: 'kIOMasterPortDefault' was deprecated in macOS 12.0: renamed to 'kIOMainPortDefault'
        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
                                                                       ^
        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
```

### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.13.9, on macOS 14.1 23B74 darwin-arm64, locale en-IQ)
    • Flutter version 3.13.9 on channel stable at /Users/ahmedhnewa/development/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (3 weeks ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/ahmedhnewa/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_SDK_ROOT = /Users/ahmedhnewa/Library/Android/sdk
    • Java binary at: /Users/ahmedhnewa/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.14.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Users/ahmedhnewa/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Community Edition (version 2023.2.5)
    • IntelliJ at /Users/ahmedhnewa/Applications/IntelliJ IDEA Community Edition.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 232.10072.19

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • Ahmed’s iPhone (mobile) • 00008020-001B59840CD8003A • ios            • iOS 17.0.3 21A360
    • macOS (desktop)         • macos                     • darwin-arm64   • macOS 14.1 23B74 darwin-arm64
    • Chrome (web)            • chrome                    • web-javascript • Google Chrome 110.0.5481.177

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Checklist before submitting a bug

- [X] I searched issues in this repository and couldn't find such bug/problem
- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

### PR DESCRIPTION
**Pull request by freshtechtips**
## Description

Fix macOS warning for device_info_plus

## Related Issues

- *Fix #2376*

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by freshtechtips** [comment]
image
I didn't get any build failure in my local machine

File: packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift [hunk 1]
```diff
@@ -4,7 +4,7 @@ public struct SystemUUID {
     
     public static func getSystemUUID() -> String? {
         let dev = IOServiceMatching("IOPlatformExpertDevice")
-        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
+        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMainPortDefault, dev)
```
Comment (ThexXTURBOXx) [inline packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift:7]: Your current implementation will break devices running older versions of macOS (as seen in the [tests](https://github.com/fluttercommunity/plus_plugins/actions/runs/6842421579/job/18603873438?pr=2377#step:6:45)).
Instead, use something like this (not sure if it works directly, you might need to fix syntax errors):

```suggestion
        if #available(macOS 12, *) {
            let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMainPortDefault, dev)
        } else {
            let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
        }
```

**Review by ThexXTURBOXx — CHANGES_REQUESTED** [review]

File: packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift (hunk 1 above)
Comment (freshtechtips) [reply packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift]: > Your current implementation will break devices running older versions of macOS (as seen in the [tests](https://github.com/fluttercommunity/plus_plugins/actions/runs/6842421579/job/18603873438?pr=2377#step:6:45)). Instead, use something like this (not sure if it works directly, you might need to fix syntax errors):

Yes that is correct.

**Review by ThexXTURBOXx — APPROVED** [review]

**Comment by freshtechtips** [comment]
I think we are ready to merge or reject, I didn't touch the dart code, for example, or even the Android Kotlin or java code, the changes are made only to swift code and this failure it might because the CI test runs from the latest YML file and it has fixes made for reverting kotlin_version and mine doesn't because I didn't touch it or it might be outdated

File: packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift [hunk 2]
```diff
@@ -4,7 +4,13 @@ public struct SystemUUID {
...
     public static func getSystemUUID() -> String? {
         let dev = IOServiceMatching("IOPlatformExpertDevice")
-        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
+        var platformExpert: io_service_t
+
+        if #available(macOS 12, *) {
+            platformExpert = IOServiceGetMatchingService(kIOMainPortDefault, dev)
+        } else {
+            platformExpert = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
+        }
```
Comment (nohli) [inline packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift:13]: Thanks for your contribution 💪🏽

Nit: Could you change the blank lines, so the section regarding platformExpert looks more belonging together?

Comment (freshtechtips) [reply packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift:13]: > Thanks for your contribution 💪🏽
> 
> Nit: Could you change the blank lines, so the section regarding platformExpert looks more belonging together?

Thanks. I would like to do what you want but unfortunately can't right now, I will do it yesterday

**Comment by freshtechtips** [comment]
> I think we are ready to merge or reject, I didn't touch the dart code, for example, or even the Android Kotlin or Java code, the changes are made only to swift code and this failure might be because the CI test runs from the latest YML file and it has fixes made for reverting kotlin_version and mine doesn't because I didn't touch it or it might be outdated

Looks like my guess was right, because it fixed by fetching the changes from the upstream remote

File: packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift (hunk 2 above)
Comment (vbuberen) [reply packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift]: Updated this part myself. Thanks for contribution again.

**Review by vbuberen — APPROVED** [review]

File: packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift (hunk 2 above)
Comment (freshtechtips) [reply packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift]: > Updated this part myself. Thanks for contribution again.

My apologies. The lack of time prevented me from doing this.

**Pull request merged**